src/promptflow/promptflow/_cli/_pf/_connection.py

```diff
@@ -194,13 +201,16 @@ def list_connection():
     print(json.dumps([connection._to_dict() for connection in connections], indent=4))
 
 
-def _upsert_connection_from_file(file, params_override=None):
+# connection_spec is used for verifying fields of custom strong type connection during the process of loading a schema.
+# Check \promptflow\src\promptflow\promptflow\_sdk\schemas\_connection.py::CustomStrongTypeConnectionSchema
+def _upsert_connection_from_file(file, params_override=None, connection_spec=None):
     # Note: This function is used for pfutil, do not edit it.
     params_override = params_override or []
     params_override.append(load_yaml(file))
-    connection = load_connection(source=file, params_override=params_override)
+    connection = load_connection(source=file, params_override=params_override, connection_spec=connection_spec)
     existing_connection = _client.connections.get(connection.name, raise_error=False)
     if existing_connection:
+        check_custom_connection_type_match(existing_connection, connection)
```

Comment:
If user call pf connection update with CLI, it will go to the 'update_connection' func below, shall we need to validate there?

Reply:
Temporarily for this PR, do not check type convert